apps/src/studio/Item.js

```diff
@@ -270,6 +283,68 @@ Item.prototype.update = function () {
 };
 
 /**
+ * Isolated update logic for "watchActor" activity where the "item" keeps
+ * turning to look at the actor with the given sprite index.
+ * @param {number} targetSpriteIndex
+ */
+Item.prototype.turnToFaceActor = function (targetSpriteIndex) {
+  // Pick a target direction closest to the relative direction toward the target.
+  var target = Studio.sprite[targetSpriteIndex];
+  if (!target) {
+    return;
+  }
+
+  // Actor positions are the top-left of their square (or their "feet" square
+  // in the 'isometric' case) - we should look at the middle of their square
+  var actorGroundCenterX = target.displayX + Studio.HALF_SQUARE;
+  var actorGroundCenterY = target.displayY + Studio.HALF_SQUARE;
+  var deltaX = actorGroundCenterX - this.x;
+  var deltaY = actorGroundCenterY - this.y;
+
+  // We shouldn't adjust our direction if the actor is sufficiently close that
+  // relative direction doesn't make much sense
+  // Basically, avoid thrashing when moving into their space.
+  var SQUARED_MINIMUM_DISTANCE = 25;
+  if (deltaX * deltaX + deltaY * deltaY > SQUARED_MINIMUM_DISTANCE) {
+    Studio.drawDebugLine("watchActor", this.x, this.y, actorGroundCenterX, actorGroundCenterY, '#ffff00');
+    this.dir = getClosestDirection(deltaX, deltaY);
+  }
+};
+
+/**
+ * Given a 2D vector (x and y) provides the closest animation direction
+ * given in our Direction enum.
+ * @param {number} x
+ * @param {number} y
+ * @returns {Direction}
+ */
+function getClosestDirection(x, y) {
+  // Y is inverted between our playlab coordinate space and what atan2 expects.
+  var radiansFromNorth = Math.atan2(x, -y);
+  var stepRadians = Math.PI / 4;
+  // Snap positive index of nearest 45° where 0 is North, 1 is NE, etc...
+  var stepsFromNorth = (Math.round(radiansFromNorth / stepRadians) + 8) % 8;
+  // At this point we should have an int between 0 and 7
+  return CLOCKWISE_DIRECTIONS_FROM_NORTH[stepsFromNorth];
+}
+
+/**
+ * Mapping number of steps away from north to direction enum.
+ * Can be indexed backwards with Array#slice.
+ * @type {Direction[]}
+ */
+var CLOCKWISE_DIRECTIONS_FROM_NORTH = [
```

Comment:
Should I move `getClosestDirection` into constants.js as well?  It's closely related, and static.